src/main/scala/uk/gov/nationalarchives/DynamoFormatters.scala

```diff
@@ -0,0 +1,151 @@
+package uk.gov.nationalarchives
+
+import cats.data._
+import cats.implicits._
+import org.scanamo._
+import org.scanamo.generic.semiauto._
+
+import java.util.UUID
+import scala.jdk.CollectionConverters._
+import scala.util.Try
+
+object DynamoFormatters {
+
+  val batchId = "batchId"
+  val id = "id"
+  val name = "name"
+  val typeField = "type"
+  val fileSize = "fileSize"
+  val sortOrder = "sortOrder"
+  val parentPath = "parentPath"
+  val title = "title"
+  val description = "description"
+  val checksumSha256 = "checksumSha256"
+  val fileExtension = "fileExtension"
+
+  private type InvalidProperty = (String, DynamoReadError)
+
+  private def stringToType(potentialTypeString: Option[String]): ValidatedNel[InvalidProperty, Type] =
+    potentialTypeString match {
+      case Some("ArchiveFolder") => ArchiveFolder.validNel
+      case Some("ContentFolder") => ContentFolder.validNel
+      case Some("Asset")         => Asset.validNel
+      case Some("File")          => File.validNel
+      case Some(typeString) => (typeField -> TypeCoercionError(new Exception(s"Type $typeString not found"))).invalidNel
+      case None             => (typeField -> MissingProperty).invalidNel
+    }
+
+  implicit val dynamoTableFormat: DynamoFormat[DynamoTable] = new DynamoFormat[DynamoTable] {
+    override def read(dynamoValue: DynamoValue): Either[DynamoReadError, DynamoTable] = {
+      val folderRowAsMap = dynamoValue.toAttributeValue.m().asScala
+
+      def readNumber[T](name: String, toNumberFunction: String => T): ValidatedNel[InvalidProperty, Option[T]] = {
```

Comment:
Maybe `readNumber`, `readValueAsNumber`, `getValueAsNumber`, `readPotentialNumber`, `readPotentialNumberValue`,`getPotentialNumber`, `getPotentialNumberValue`

Reply:
I'll go with `getPotentialNumber` as it matches `getPotentialStringValue`